### PR TITLE
[SYCL] Fix -fno-sycl-early-optimizations behavior broken in pulldown.

### PR DESCRIPTION
**Pull request by tianqingw**
D83892 simplifies Clang option parsing, including that of
"-disable-llvm-passes", but inadvertently removes SYCL custom logic for it.
This patch reverts to the old way of handling "-disable-llvm-passes" and
restores the original behavior.

### CODE REVIEW

**Comment by elizabethandrews** [comment]
Should this have tests? Or is this just fixing failing tests because of incorrect pulldown?

**Comment by tianqingw** [comment]
> Should this have tests? Or is this just fixing failing tests because of incorrect pulldown?

Yeah, I forgot to add test.

**Review by premanandrao — APPROVED** [review]
Looks okay to me.  @elizabethandrews?

**Review by mdtoguchi — APPROVED** [review]

**Review by elizabethandrews — APPROVED** [review]
LGTM. Just had one question

File: clang/test/CodeGenSYCL/kernel-early-optimization-pipeline.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Check LLVM optimization pipeline is run by default for SPIR-V compiled for
+// SYCL device target, and can be disabled with -fno-sycl-early-optimizations.
+//
+// RUN: %clang_cc1 -O2 -fsycl -fsycl-is-device -triple spir64-unknown-unknown-sycldevice %s -mllvm -debug-pass=Structure -emit-llvm -o - 2>&1 | FileCheck %s --check-prefix=CHECK-EARLYOPT
+// CHECK-EARLYOPT: Lower Work Group Scope Code
+// CHECK-EARLYOPT: Combine redundant instructions
+//
+// RUN: %clang_cc1 -O2 -fsycl -fsycl-is-device -fno-sycl-early-optimizations -triple spir64-unknown-unknown-sycldevice %s -mllvm -debug-pass=Structure -emit-llvm -o - 2>&1 | FileCheck %s --check-prefix=CHECK-NOEARLYOPT
+// CHECK-NOEARLYOPT: Lower Work Group Scope Code
```
[the remark this answers is not in the document]
Comment (secondwtq) [reply clang/test/CodeGenSYCL/kernel-early-optimization-pipeline.cpp:9]: Yes, this one is separate from the others:
https://github.com/intel/llvm/blob/051e1407a1951e32274564656847cc6335020ac6/clang/lib/CodeGen/CodeGenAction.cpp#L347

**Comment by bader** [comment]
FYI: D83892 is reverted by https://reviews.llvm.org/rG7ad666798f12456d9e663e763e17e29007c3728d (Dec 23) and re-applied by https://reviews.llvm.org/rGce8c59e6af487f0b8786ae921aa926341f0ae04f (Jan 5).

**Comment by bader** [comment]
I suggest we integrate this fix to #3015, which includes the final version of the community change.

**Comment by bader** [comment]
I see that this fix is included into #3015, so I'm closing this one.

**Pull request closed without merging**